python/__init__.py

```diff
@@ -95,6 +99,8 @@ def unpack_can_buffer(dat):
         break
   return ret
 
+# TODO: type hints for these decorators
```

Comment:
i'm not exactly sure how to type these correctly, so i've left them out for now.
i'll figure them out and either add to this PR, or submit it in another one.

i think it's something along the lines of
```py
from typing import TypeVar
from typing_extension import ParamSpec

P = ParamSpec("P")
R = TypeVar("R")


def ensure_health_packet_version(fn: Callable[Concatenate[Panda, P], R]) -> Callable[Concatenate[Panda, P], R]:
  @wraps(fn)
  def wrapper(self, *args: P.args, **kwargs: P.kwargs) -> R:
    if self.health_version < self.HEALTH_PACKET_VERSION:
      raise RuntimeError("Panda firmware has outdated health packet definition. Reflash panda firmware.")
    elif self.health_version > self.HEALTH_PACKET_VERSION:
      raise RuntimeError("Panda python library has outdated health packet definition. Update panda python library.")
    return fn(self, *args, **kwargs)
  return wrapper

def ensure_can_packet_version(fn: Callable[Concatenate[Panda, P], R]) -> Callable[Concatenate[Panda, P], R]:
  @wraps(fn)
  def wrapper(self, *args: P.args, **kwargs: P.kwargs) -> R:
    if self.can_version < self.CAN_PACKET_VERSION:
      raise RuntimeError("Panda firmware has outdated CAN packet definition. Reflash panda firmware.")
    elif self.can_version > self.CAN_PACKET_VERSION:
      raise RuntimeError("Panda python library has outdated CAN packet definition. Update panda python library.")
    return fn(self, *args, **kwargs)
  return wrapper
```